pkg/apis/acme/v1/types_order.go

```diff
@@ -69,7 +69,14 @@ type OrderSpec struct {
 	// DNSNames is a list of DNS names that should be included as part of the Order
 	// validation process.
 	// This field must match the corresponding field on the DER encoded CSR.
-	DNSNames []string `json:"dnsNames"`
+	//+optonal
+	DNSNames []string `json:"dnsNames,omitempty"`
+
+	// IPAddresses is a list of IP addresses that should be included as part of the Order
+	// validation process.
+	// This field must match the corresponding field on the DER encoded CSR.
+	// +optional
+	IPAddresses []string `json:"ipAddresses,omitempty"`
```

Comment:
Nit: Also update the documentation for `commonName` which currently says:
> If specified, this value must also be present in `dnsNames`.

To

> If specified, this value must also be present in `dnsNames` or `ipAddresses`.